plugins/scenario/hu.bme.mit.gamma.scenario.statechart.util/src/hu/bme/mit/gamma/scenario/statechart/util/transformation/AutomatonDeterminizator.xtend

```diff
@@ -48,6 +48,8 @@ class AutomatonDeterminizator {
 		this.oldFirstRegion = oldStatechart.regions.head
 		this.newStatechart = oldStatechart.clone
 		this.firstRegion = newStatechart.regions.head
+		newStatechart.variableDeclarations.clear
+		newStatechart.variableDeclarations += oldStatechart.variableDeclarations
```

Comment:
Not sure about this one either: shouldn't these variable declarations be cloned?